### PR TITLE
Add temporary file upload

### PR DESCRIPTION
**Pull request by djfhe**

### CODE REVIEW

**Comment by djfhe** [comment]
https://user-images.githubusercontent.com/44621469/209334390-f1e78ad0-40b1-4c4f-84a8-9bb56796c8ab.mp4

File: application/src/jvmMain/kotlin/replace/usecase/file/DeleteFileUserCase.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package replace.usecase.file
+
+import org.bson.types.ObjectId
+import replace.datastore.FileRepository
+import replace.datastore.FileStorage
+
+object DeleteFileUserCase {
```
Comment (alexstaeding) [inline application/src/jvmMain/kotlin/replace/usecase/file/DeleteFileUserCase.kt:7]: UserCase -> UseCase

File: application/src/jvmMain/kotlin/replace/usecase/temporaryfileupload/CreateTemporaryFileUploadUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+object CreateTemporaryFileUploadUseCase {
+
+    suspend fun execute(
+        fileName: String,
+        input: InputStream,
+        temporaryFileRepository: TemporaryFileRepository,
+        fileStorage: FileStorage
+    ): TemporaryFileUploadDto {
+
+        val temporaryFileUploadPath = "temporary_uploads/${java.util.UUID.randomUUID()}"
```
Comment (alexstaeding) [inline application/src/jvmMain/kotlin/replace/usecase/temporaryfileupload/CreateTemporaryFileUploadUseCase.kt:20]: Replace FQ name with import

File: application/src/jvmMain/kotlin/replace/usecase/temporaryfileupload/CreateTemporaryFileUploadUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import java.io.InputStream
+import java.net.URLConnection
+
+object CreateTemporaryFileUploadUseCase {
+
+    suspend fun execute(
+        fileName: String,
+        input: InputStream,
+        temporaryFileRepository: TemporaryFileRepository,
+        fileStorage: FileStorage
```
Comment (alexstaeding) [inline application/src/jvmMain/kotlin/replace/usecase/temporaryfileupload/CreateTemporaryFileUploadUseCase.kt:17]: Trailing comma

File: application/src/jvmMain/kotlin/replace/dto/FileDto.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package replace.dto
+
+import kotlinx.serialization.Serializable
+import replace.model.File
+
+@Serializable
+class FileDto(
```
Comment (alexstaeding) [inline application/src/jvmMain/kotlin/replace/dto/FileDto.kt:7]: Put properties in same order as model class

File: application/src/jvmMain/kotlin/replace/usecase/temporaryfileupload/DeleteTemporaryFileUploadUseCase.kt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+object DeleteTemporaryFileUploadUseCase {
+
+    suspend fun execute(
+        temporaryFileUploadId: String,
+        temporaryFileRepository: TemporaryFileRepository,
+        fileStorage: FileStorage
+    ) {
+
+        if (!ObjectId.isValid(temporaryFileUploadId)) {
+            throw IllegalArgumentException("Id $temporaryFileUploadId: is not a valid ObjectId")
```
Comment (alexstaeding) [inline application/src/jvmMain/kotlin/replace/usecase/temporaryfileupload/DeleteTemporaryFileUploadUseCase.kt:16]: Without : ?

File: infrastructure/src/jvmMain/kotlin/replace/job/Job.kt [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import java.util.concurrent.Executors
+import kotlin.coroutines.CoroutineContext
+
+abstract class Job(
```
Comment (alexstaeding) [inline infrastructure/src/jvmMain/kotlin/replace/job/Job.kt:11]: Find a better name that doesn't conflict with Job from coroutines

**Review by alexstaeding — CHANGES_REQUESTED** [review]

File: application/src/jvmMain/kotlin/replace/usecase/temporaryfileupload/CreateTemporaryFileUploadUseCase.kt (hunk 3 above)
Comment (djfhe) [reply application/src/jvmMain/kotlin/replace/usecase/temporaryfileupload/CreateTemporaryFileUploadUseCase.kt]: Fixed manually, but such styling errors should be fixed automatically by ktlint when possible.

Comment (alexstaeding) [reply application/src/jvmMain/kotlin/replace/usecase/temporaryfileupload/CreateTemporaryFileUploadUseCase.kt]: https://github.com/pinterest/ktlint/issues/709

Comment (djfhe) [reply application/src/jvmMain/kotlin/replace/usecase/temporaryfileupload/CreateTemporaryFileUploadUseCase.kt]: I would add the rule directly on the master later or another branch, not in this one

**Review by alexstaeding — APPROVED** [review]
LGTM

**Pull request merged**